go/store/blobstore/gcs.go

```diff
@@ -155,3 +162,61 @@ func (bs *GCSBlobstore) CheckAndPut(ctx context.Context, expectedVersion, key st
 
 	return ver, err
 }
+
+func (bs *GCSBlobstore) Contatenate(ctx context.Context, key string, sources []string) (string, error) {
+	// GCS compose has a batch size limit,
+	// recursively compose sources
+	for len(sources) > composeBatch {
+		// compose subsets of |sources| in batches,
+		// store tmp composite objects in |next|
+		var next []string
+		for len(sources) > 0 {
+			tmp := uuid.New().String()
+			batch := min(composeBatch, len(sources))
+			_, err := bs.composeObjects(ctx, tmp, sources[:batch])
```

Comment:
We expect to call this with ~250 sources in the current code paths and common configurations?

It would be nice, but maybe not necessary, to parallelize the calls to composeObjects at each level...

```go
var next []string, batches [][]string
for len(sources > 0) {
    batchSz := min(composeBatch, len(sources))
    batches = append(batches, sources[:batchSz])
    next = append(next, uuid.New().String())
    sources = sources[batchSz:]
}
eg, ctx := errgroup.WithContext(ctx)
for i := 0; i < len(batches); i++ {
    i := i
    eg.Go(func() error {
        _, err := bs.composeObjects(ctx, next[i], batches[i])
        return err
    })
}
err := eg.Wait()
if err != nil {
    return "", err
}
sources = next